### PR TITLE
Verifying if the matrix is symmetric positive definite during Cholesky Decomposition

### PR DESCRIPTION
**Pull request by CaptainSharf**
As part of this PR,
- Adding error message if Cholesky decomposition fails for real matrices
- Adding error message if Cholesky decomposition fails for complex matrices
- Adding tests for the changed functionality

### CODE REVIEW

**Comment by CaptainSharf** [comment]
Hi @ben-albrecht ,
Could you please validate if this change I made is is valid?

File: modules/packages/LinearAlgebra.chpl [hunk 1]
```diff
@@ -1633,8 +1633,13 @@ proc cholesky(A: [] ?t, lower = true)
 
   var copy = A;
   const uploStr = if lower then "L" else "U";
-  LAPACK.potrf(lapack_memory_order.row_major, uploStr, copy);
-
+  var hasError = LAPACK.potrf(lapack_memory_order.row_major, uploStr, copy);
+  if(hasError >0){
```
Comment (ben-albrecht) [inline modules/packages/LinearAlgebra.chpl:1637]:
```suggestion
  if(hasError > 0){
```

File: test/library/packages/LinearAlgebra/correctness/testCholeskyComplex.chpl [hunk 2]
```diff
@@ -0,0 +1,8 @@
+use LinearAlgebra;
```
Comment (ben-albrecht) [inline test/library/packages/LinearAlgebra/correctness/testCholeskyComplex.chpl:1]: How about we clarify that these tests are checking an error case in the file name, e.g. `testCholeskyComplexError.{chpl,good}` and `testCholeskyRealError.{chpl,good}`

**Review by ben-albrecht — COMMENTED** [review]
@CaptainSharf - This looks great!

I have some minor suggestions inline. Other than that, give us some time to process your CLA before we are able to merge this.

Thanks!

**Review by ben-albrecht — APPROVED** [review]

**Comment by ben-albrecht** [comment]
Thanks @CaptainSharf !

**Pull request merged**